app/models/todos/undone_todos_query.rb

```diff
@@ -27,6 +27,9 @@ def query(params)
 
       if params[:tag]
         tag = Tag.where(:name => params[:tag]).first
+       if not tag
```

Comment:
Would you mind flattening this out a bit?
```suggestion
       return [] if !tag
```

The english `and`, `or`, and `not` in Ruby behave differently from `&&`, `||`, and `!` respectively and for comparisons like this, I'd like us to stick to the non-english version.